sda/service.go

```diff
@@ -189,37 +199,74 @@ func newServiceStore(h *Host, o *Overlay) *serviceStore {
 		if err := os.MkdirAll(configName, 0666); err != nil {
 			log.Error("Service", name, "Might not work properly: error setting up its config directory(", configName, "):", err)
 		}
-		c := newContext(h, o, id)
+		c := newContext(h, o, id, s)
 		s, err := ServiceFactory.start(name, c, configName)
 		if err != nil {
 			log.Error("Trying to instantiate service:", err)
 		}
 		log.Lvl2("Started Service", name, " (config in", configName, ")")
 		services[id] = s
 		configs[id] = configName
-		// !! register to the ProtocolFactory !!
-		//ProtocolFactory.registerService(id, s.NewProtocol)
 	}
 	log.Lvl3(h.workingAddress, "instantiated all services")
-	return &serviceStore{services, configs}
+
+	// registering messages that services are expecting
+	h.RegisterProcessor(s, RequestID)
+	return s
+}
+
+// Process implements the Processor interface: service manager will relay
+// messages to the right Service.
+func (s *serviceManager) Process(data *network.Packet) {
+	id := data.ServerIdentity
+	switch data.MsgType {
+	case RequestID:
+		r := data.Msg.(ClientRequest)
+		// check if the target service is indeed existing
+		s, ok := s.serviceByID(r.Service)
+		if !ok {
+			log.Error("Received a request for an unknown service", r.Service)
+			// XXX TODO should reply with some generic response =>
+			// 404 Service Unknown
+			return
+		}
+		go s.ProcessClientRequest(id, &r)
+	default:
+		// will launch a go routine for that message
+		s.Dispatch(data)
+	}
+}
+
+// Register the processor to the service manager and tells the host to dispatch
```

Comment:
Maybe because it recognizes it implements an `Interface` and then he does not require to put any comments on top (because godoc could re-use the comment of the `Interface` ?)